R/read_file.R

```diff
@@ -0,0 +1,80 @@
+#' Read files with extension .fst, .csv, and .rda (.rds and .RData)
+#'
+#' @param file_name Character; path to a file storing a single object.
+#'
+#' @noRd
+#' @examples
+#' # Odd case: .RData saved with extension .rda
+#' path <- tempfile(fileext = ".rda")
+#'
+#' data1 <- data.frame(x = 1)
+#' saveRDS(data1, path)
+#' read_file(path)
+#'
+#' # More natural case: .RData saved as .rda
+#' data2 <- data.frame(x = 1)
+#' save(data2, file = path)
+#' read_file(path)
+#'
+#' # Fails by design: .RData with more than 1 object
+#'
+#' save(data1, data2, file = path)
+#' # Fails
+#' try(read_file(path))
+read_file <- function(path) {
+  # FIXME: A warning seems like a bad idea. It should likely be an error
+  if (!fs::file_exists(path)) {
+    rlang::warn("The `path` does not exist: {path}")
+    return(invisible(path))
+  }
+
+  abort_unsupported_extension(path)
+
+  extension <- fs::path_ext(path)
+  switch(
+    extension,
+    "rda" = read_rda(path),
```

Comment:
I think we should consider adding ".rds" as a supported extension also, that way if we ever begin the process of properly naming exported files, this function will continue to work in place, e.g. if a script uses this function to read in an RDS file that is named "data.rda" and we eventually fix the upstream code that creates "data.rda" to export that data with the proper file name/extension "data.rds" then the script could continue to function without modification.